frontend/src/features/SideWindow/MissionForm/ActionForm/shared/VesselField.tsx

```diff
@@ -0,0 +1,62 @@
+import { useFormikContext } from 'formik'
+import { useCallback, useMemo } from 'react'
+import styled from 'styled-components'
+
+import { useNewWindow } from '../../../../../ui/NewWindow'
+import { VesselSearch } from '../../../../VesselSearch'
+
+import type { VesselIdentity } from '../../../../../domain/entities/vessel/types'
+import type { MissionActionFormValues } from '../../types'
+
+export function VesselField() {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const { setFieldValue, values } = useFormikContext<MissionActionFormValues>()
+
+  const { newWindowContainerRef } = useNewWindow()
+
+  const defaultValue = useMemo(
+    () => ({
+      flagState: values.vesselFlagState,
+      vesselName: values.vesselName
+    }),
+    [values.vesselFlagState, values.vesselName]
+  )
+
+  const handleVesselChange = useCallback(
+    (nextVessel: VesselIdentity | undefined) => {
+      if (!nextVessel) {
+        setFieldValue('vesselFlagState', undefined)
```

Comment:
vesselFlagState -> flagState

Reply:
We do that for all the 5 vessel props or just for this one?:

```ts
  export interface MissionAction {
    // ...

    vesselExternalReferenceNumber: string
    vesselFlagState: string
    vesselInternalReferenceNumber: string
    vesselIrcs: string
    vesselName: string
  }
```

Reply:
externalReferenceNumber: string
    flagState: string
    internalReferenceNumber: string
    ircs: string
    vesselName: string